packages/blade/src/components/Switch/ThumbIcon.tsx

```diff
@@ -0,0 +1,54 @@
+import isNumber from 'lodash/isNumber';
+import type { SwitchProps } from './types';
+import { AnimatedThumbIcon } from './AnimatedThumbIcon';
+import { switchColors } from './switchTokens';
+import { useTheme } from '~components/BladeProvider';
+import { getIn, makeSize, makeSpace, useBreakpoint } from '~utils';
+import size from '~tokens/global/size';
+import { Path } from '~components/Icons/_Svg';
+
+const switchIconSize = {
+  desktop: {
+    small: size[6],
+    medium: 'spacing.3',
+  },
+  mobile: {
+    small: 'spacing.3',
+    medium: size[10],
+  },
+} as const;
+
+const ThumbIcon = ({
+  isChecked,
+  isDisabled,
+  size = 'medium',
+}: Pick<SwitchProps, 'isChecked' | 'isDisabled' | 'size'>): React.ReactElement => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
```

Comment:
Yeah so the platform will contain `onDesktop` where I need `desktop`. Can do string manipulation but looks kinda hacky.

Reply:
You need `desktop` because you've structured your tokens that way, make it `onDesktop` instead 😅 

Reason is that `platform` is cleanly exposed from useTheme and `onDesktop` & `onMobile` is a terminology we use on our global tokens so better to keep the same on component level tokens